sdk/core/azure-core/src/main/java/com/azure/core/models/HttpResponseError.java

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.models;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.List;
+
+/**
+ * This class represents the error details of an HTTP response.
+ */
+public final class HttpResponseError {
+
+    @JsonProperty(value = "code", required = true)
+    private final String code;
+
+    @JsonProperty(value = "message", required = true)
+    private final String message;
+
+    @JsonProperty(value = "target")
+    private String target;
+
+    @JsonProperty(value = "innererror")
+    private HttpResponseInnerError innerError;
+
+    @JsonProperty(value = "details")
+    private List<HttpResponseError> errorDetails;
+
+    /**
+     * Creates an instance of {@link HttpResponseError}.
+     *
+     * @param code the error code of this error.
+     * @param message the error message of this error.
+     */
+    @JsonCreator
+    public HttpResponseError(@JsonProperty(value = "code", required = true)String code,
+                             @JsonProperty(value = "message", required = true)String message) {
+        this.code = code;
+        this.message = message;
+    }
+
+    /**
+     * Returns the error code of this error.
+     *
+     * @return the error code of this error.
+     */
+    public String getCode() {
+        return code;
+    }
+
+    /**
+     * Returns the error message of this error.
+     *
+     * @return the error message of this error.
+     */
+    public String getMessage() {
+        return message;
+    }
+
+    /**
+     * Returns the target of this error.
+     *
+     * @return the target of this error.
+     */
+    public String getTarget() {
+        return target;
+    }
+
+    /**
+     * Sets the target of this error.
+     *
+     * @param target the target of this error.
+     * @return the updated {@link HttpResponseError} instance.
+     */
+    public HttpResponseError setTarget(String target) {
```

Comment:
Because this is only being consumed by end users (and not created), do we really need setters anywhere?

Reply:
Since this type is in core and used by client libraries, I included ctor args and setters to differentiate between required fields and optional fields. If not, we'll have to provide multiple overloads with different optional fields combinations.